### PR TITLE
Add some explanation to application_services.md

### PR DESCRIPTION
**Pull request by lub**
Signed-off-by: Simon Körner <git@lubiland.de>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [ ] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

This PR does not touch code, are linters and changelog still neccessary?

### CODE REVIEW

File: docs/application_services.md [hunk 1]
```diff
@@ -23,9 +23,13 @@ namespaces:
   users:  # List of users we're interested in
     - exclusive: <bool>
       regex: <regex>
+      group_id: <group>
     - ...
   aliases: []  # List of aliases we're interested in
   rooms: [] # List of room ids we're interested in
 ```
 
+`exclusive`: If enabled registrations of users in that namespace are only allowed for the application service
```
Comment (anoadragon453) [inline docs/application_services.md]:
```suggestion
`exclusive`: If enabled, only this application service is allowed to register users in that namespace
```

File: docs/application_services.md [hunk 2]
```diff
@@ -23,9 +23,13 @@ namespaces:
...
     - exclusive: <bool>
       regex: <regex>
+      group_id: <group>
     - ...
   aliases: []  # List of aliases we're interested in
   rooms: [] # List of room ids we're interested in
 ```
 
+`exclusive`: If enabled registrations of users in that namespace are only allowed for the application service  
+`group_id`: All users of this application service are dynamically registered for this group to e.g. use it as flair
```
Comment (anoadragon453) [inline docs/application_services.md]:
```suggestion
`group_id`: All users of this application service are dynamically joined to this group. This is useful for e.g user organisation or flairs.
```

**Comment by clokep** [comment]
I went ahead and added the newsfragement file since this seems like something to push over the finish line. @anoadragon453 does this still look good?

File: changelog.d/7091.doc [hunk 3]
```diff
@@ -0,0 +1 @@
+Improve the documentation of application services configuration files.
```
Comment (anoadragon453) [inline changelog.d/7091.doc:1]:
```suggestion
Improve the documentation of application service configuration files.
```

File: docs/application_services.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ namespaces:
   rooms: [] # List of room ids we're interested in
 ```
 
-`exclusive`: If enabled registrations of users in that namespace are only allowed for the application service  
-`group_id`: All users of this application service are dynamically registered for this group to e.g. use it as flair
+`exclusive`: If enabled, only this application service is allowed to register users in that namespace
```
Comment (anoadragon453) [inline docs/application_services.md:32]:
```suggestion
`exclusive`: If enabled, only this application service is allowed to register users in its namespace(s).
```

**Pull request merged**